Sources/StreamChat/Utils/MessagesPaginationStateHandling/MessagesPaginationStateHandling.swift

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A component responsible for handling the messages pagination state.
+protocol MessagesPaginationStateHandling {
```

Comment:
You are subclassing it in `MessagesPaginationStateHandler_Mock`. The protocol so far has no benefit. It may have in the future but given that (at least I don't have enough data) it's not really obvious how this is helping here.

Reply:
`MessagesPaginationStateHandler_Mock` is not a subclass, it is an implementation of the protocol `MessagesPaginationStateHandling`. 

As I said before, it benefits the unit tests. I know people normally say "Do not write code for the sake of tests" (I've been there as well), but the reality is that it contributes to the developer experience. With the current approach, I'm able to write the following unit tests:
```swift
func test_begin_whenLoadingFirstPage_thenSetsStateToInitial() {
    // Given
    sut.state.isLoadingNextMessages = true
    sut.state.hasLoadedAllNextMessages = false
    sut.state.oldestFetchedMessage = .dummy()
    let pagination = MessagesPagination(pageSize: 10, parameter: nil)

    // When
    sut.begin(pagination: pagination)

    // Then
    XCTAssertEqual(sut.state, .initial)
}
```

Without the protocol, and because the `state` property would have to be `private(set)`, I would have to write them like this:
```swift
func test_begin_whenLoadingFirstPage_thenSetsStateToInitial() {
    // Given
    // simulate I already did load another page before
    let pagination = MessagesPagination(pageSize: 5, parameter: .around("122"))
    let messages: [MessagePayload] = [
        .dummy(messageId: "121"),
        .dummy(messageId: "122"),
        .dummy(messageId: "123")
    ]
    sut.end(pagination: pagination, with: .success(messages))

    // When
    let pagination = MessagesPagination(pageSize: 10, parameter: nil)
    sut.begin(pagination: pagination)

    // Then
    XCTAssertEqual(sut.state, .initial)
}
```
As you can see, it is much harder to read and understand what is happening, besides that is not really a "unit" test since it calls 2 public functions.

Reply:
I totally agree, I misread and thought that that was a subclass.